sdk/python/feast/feature_server.py

```diff
@@ -222,8 +223,12 @@ def write_to_online_store(body=Depends(get_body)):
         )
 
     @app.get("/health")
-    def health():
-        return Response(status_code=status.HTTP_200_OK)
+    async def health():
+        return (
+            Response(status_code=status.HTTP_200_OK)
+            if registry_proto
+            else Response(status_code=status.HTTP_503_SERVICE_UNAVAILABLE)
+        )
```

Comment:
if a path operation can be async in fastapi, it should be. update it to reflect that without a registry, the service is unavailable.